### PR TITLE
Optimization loop chassis

### PR DESCRIPTION
**Pull request by akilandrews**
Updates include:
- Scaling training features using learned length scales
- Support for reusing optimizer instance
- Added two parameters: keep_state and probe_previous
- Update BayesianOptimization.TargetSpace.target_function with new objective function

Evaluated performance using 2D anisotropic distortion distance function (simple case) experimenting with four treatments:
- 1 (baseline) - brute force workflow following Anisotropic Metric Tutorial using single optimization step.
- 2a (opt loop) each loop iteration create new optimizer instance, initialize with new objective function, and do not probe previous points 
- 2b (opt loop) each loop iteration create new optimizer instance, initialize with new objective function, and probe previous points 
- 2c (opt loop) each loop iteration reuse optimizer instance, set target_function to new objective function, and do not probe previous points

### CODE REVIEW

**Review by bwpriest — CHANGES_REQUESTED** [review]
You can hook `tests/mini_batch.py` back up to the CI harness by uncommenting the [these lines](https://github.com/akilandrews/MuyGPyS/blob/def6effea5c597051d89e7bd46dbb667bbfe4f58/.github/workflows/develop-test.yml#L49-L51) and adding `optimize-experimental` to the list of test groups [here](https://github.com/akilandrews/MuyGPyS/blob/def6effea5c597051d89e7bd46dbb667bbfe4f58/.github/workflows/develop-test.yml#L17). I think it will then be ready to merge!

**Comment by bwpriest** [comment]
Don't worry about occasional failures in the `3.X optimize` checks. You are did not do anything wrong. Occasionally the number of BO probes is insufficient to converge to a good value or the sampler produces a particularly yucky surface. I am using generate tolerances and several iterations, but sometimes there is still a failure. This is a classic problem associated with testing stochastic code.

**Review by bwpriest — CHANGES_REQUESTED** [review]
Sorry, I cut in line and merged a PR ahead of this because I found and fixed a major bug in how the (anisotropic at least) models get initialized after optimization. You might notice changes in your codes. Hopefully the changes are positive. #182 also changes how the sampler chooses the training data - so there are some minor changes to the `UnivariateSampler2D` API. You'll need to update your codes that use the `UnivariateSampler2D`, but it should be very easy to do so. Check out the anisotropic tutorial for an example (note the changes in the sampled training points).

**Review by bwpriest — APPROVED** [review]
LGTM
